lib/git-fastclone.rb

```diff
@@ -217,19 +209,16 @@ def clone(url, rev, src_dir, config)
       with_git_mirror(url) do |mirror, attempt_number|
         clear_clone_dest_if_needed(attempt_number, clone_dest)
 
-        clone_command = '--quiet --reference :mirror :url :path'
-        clone_command += ' --config :config' unless config.nil?
-        Terrapin::CommandLine.new('git clone', clone_command)
-                             .run(mirror: mirror.to_s,
-                                  url: url.to_s,
-                                  path: clone_dest,
-                                  config: config.to_s)
+        clone_commands = ['git', 'clone', verbose ? '--verbose' : '--quiet']
+        clone_commands << "--reference" << mirror.to_s << url.to_s << clone_dest
+        clone_commands << "--config" << config.to_s unless config.nil?
+        fail_pipe_on_error(clone_commands, quiet: !verbose)
       end
 
       # Only checkout if we're changing branches to a non-default branch
       if rev
         Dir.chdir(File.join(abs_clone_path, src_dir)) do
-          Terrapin::CommandLine.new('git checkout', '--quiet :rev').run(rev: rev.to_s)
+          fail_pipe_on_error(['git', 'checkout', '--quiet', rev.to_s], quiet: !verbose)
```

Comment:
Why use `fail_pipe_on_error` here and below? Doesn't look like the input of one command is being piped into another AFAICT.

Reply:
discussed offline: fail_pipe_on_error also helps streaming the output to STDOUT when we set verbose to true